video/out/vo_opengl_cb.c

```diff
@@ -171,16 +172,22 @@ int mpv_opengl_cb_init_gl(struct mpv_opengl_cb_context *ctx, const char *exts,
         return MPV_ERROR_UNSUPPORTED;
     }
 
-    ctx->ra = ra_create_gl(ctx->gl, ctx->log);
-    if (!ctx->ra)
+    // initialize a blank ra_ctx to reuse ra_gl_ctx
+    ctx->ra_ctx = talloc_zero(ctx, struct ra_ctx);
+    ctx->ra_ctx->log = ctx->log;
+    ctx->ra_ctx->global = ctx->global;
+
+    // .external_swchain is set to true since we never call swap_buffers
+    struct ra_gl_ctx_params gl_params = { .external_swchain = true };
+    if (!ra_gl_ctx_init(ctx->ra_ctx, ctx->gl, gl_params))
         return MPV_ERROR_UNSUPPORTED;
```

Comment:
It shouldn't error out if a software renderer is detected. It shouldn't mess with SwapInterval at all. I suspect the get_native_display stuff is wrong too, but I don't have the overview right now.